scripts/vm/hypervisor/kvm/kvmheartbeat_rbd.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+help() {
+  printf "Usage: $0 
+                    -p rbd pool name
+                    -n pool auth username
+                    -s pool auth secret
+                    -h host
+                    -i source host ip
+                    -r cretae/read hb watcher
+                    -c cleanup"
+  exit 1
+}
+#set -x
+PoolName=
+PoolAuthUserName=
+PoolAuthSecret=
+HostIP=
+SourceHostIP=
+rflag=0
+cflag=0
+
+while getopts 'p:n:s:h:i:rc' OPTION
+do
+  case $OPTION in
+  p)
+     PoolName="$OPTARG"
+     ;;
+  n)
+     PoolAuthUserName="$OPTARG"
+     ;;
+  s)
+     PoolAuthSecret="$OPTARG"
+     ;;
+  h)
+     HostIP="$OPTARG"
+     ;;
+  i)
+     SourceHostIP="$OPTARG"
+     ;;
+  r)
+     rflag=1 
+     ;;
+  c)
+    cflag=1
+     ;;
+  *)
+     help
+     ;;
+  esac
+done
+
+if [ -z "$PoolName" ]; then
+  exit 2
+fi
+
+keyringFile="/etc/cloudstack/agent/keyring"
```

Comment:
This will break if multiple Ceph clusters are attached to the same host.

This is technically possible and also done (we do this for example).

At least you need a unique keyfile per Ceph cluster, but I would even say that we should not write this credentials to a file anyway.

If we do so at least write it to a temporary directory in /run somewhere and make sure the chmod permissions are set properly so no other user can read it.